tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/NLog/LogsInjection/ScopeContextSetterProxy.cs

```diff
@@ -0,0 +1,27 @@
+// <copyright file="ScopeContextSetterProxy.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.LogsInjection
+{
+    /// <summary>
+    /// Duck type for ScopeContextSetterProxy in NLog 5.0+
+    /// </summary>
+    internal class ScopeContextSetterProxy
+    {
+        /// <summary>
+        /// Updates the logical scope context with provided properties
+        /// </summary>
+        /// <param name="properties">Properties being added to the scope dictionary</param>
+        /// <returns>A disposable object that removes the properties from logical context scope on dispose.</returns>
+        /// <remarks>Scope dictionary keys are case-insensitive</remarks>
+        public virtual IDisposable PushProperties(IReadOnlyList<KeyValuePair<string, object>> properties)
+        {
+            return null;
+        }
+    }
```

Comment:
Why a class and not an interface for the proxy definition?

Reply:
Class and Abstract class proxy definitions solves some corner cases, but for the 98% of case an interface or a duckcopy struct will do the job.

There's a "Best Practices" section in the docs: https://github.com/DataDog/dd-trace-dotnet/blob/master/docs/development/DuckTyping.md#best-practices

Reply:
It was just out of copying what the MDC and MDLC ones were doing.

Thanks for that link, can you give me some clarification on this sentence? `ISomeObject as long as they can be used in a constraint` what is a constraint in this context?

Reply:
`constraint` in this case refers to the method constraints in the CallTarget integration. For example: https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/HttpClient/HttpMessageHandlerCommon.cs#L20

Reply:
the trick here is because is a generic argument we don't box the struct type, and with the constraint you can call interface members for that generic proxy type instance.